src/components/settings/PresetButtons.svelte

```diff
@@ -0,0 +1,136 @@
+  let active = $activePreset;
+  $: activePreset.set(active);
+
+  async function updatePreset(number: number) {
+    if (number !== active) {
+      importStores(JSON.stringify($presets[number - 1]));
+      active = number;
+      return;
+    }
+
+    const presetData = $presets;
+    presetData[number - 1] = {
+      captionDuration: $captionDuration,
+      captionFontSize: $captionFontSize,
+      captionWidth: $captionWidth,
+      captionLeft: $captionLeft,
+      captionTop: $captionTop,
+
+      videoSideSetting: $videoSideSetting,
+      videoPanelSize: $videoPanelSize,
+
+      chatZoom: $chatZoom,
+      chatSize: $chatSize,
+      chatSplit: $chatSplit,
+
+      textDirection: $textDirection,
+      livetlFontSize: $livetlFontSize,
+
+      keyboardShortcuts: $keyboardShortcuts,
+
+      doTranslatorMode: $doTranslatorMode,
+      doAutoPrefix: $doAutoPrefix,
+      autoPrefixTag: $autoPrefixTag,
+      macroTrigger: $macroTrigger,
+      macros: $macros
+    };
+    presets.set(presetData);
+  }
+
+  async function deletePreset(number: number) {
+    const presetData = $presets;
+    presetData.splice(number - 1, 1);
+    presets.set(presetData);
+    presetCount -= 1;
+  }
+
+  async function addPreset() {
+    const presetData = $presets;
+    presetData.push({
+      captionDuration: captionDuration.defaultValue,
+      captionFontSize: captionFontSize.defaultValue,
+      captionWidth: captionWidth.defaultValue,
+      captionLeft: captionLeft.defaultValue,
+      captionTop: captionTop.defaultValue,
+
+      videoSideSetting: videoSideSetting.defaultValue,
+      videoPanelSize: videoPanelSize.defaultValue,
+
+      chatZoom: chatZoom.defaultValue,
+      chatSize: chatSize.defaultValue,
+      chatSplit: chatSplit.defaultValue,
+
+      textDirection: textDirection.defaultValue,
+      livetlFontSize: livetlFontSize.defaultValue,
+
+      keyboardShortcuts: keyboardShortcuts.defaultValue,
+
+      doTranslatorMode: doTranslatorMode.defaultValue,
+      doAutoPrefix: doAutoPrefix.defaultValue,
+      autoPrefixTag: autoPrefixTag.defaultValue,
+      macroTrigger: macroTrigger.defaultValue,
+      macros: macros.defaultValue
+    });
+    presets.set(presetData);
+    presetCount += 1;
+  }
+</script>
+
+{#each getPresetArray(presetCount) as i} 
+  <div class="flex gap-2 py-1">
+    <Button add="flex-1" on:click={() => { isDeleting ? deletePreset(2 * i - 1) : updatePreset(2 * i - 1); }} color="dark" light={active === 2 * i - 1 && !isDeleting}>
+      {isDeleting ? `Delete Preset ${2 * i - 1}` : active !== 2 * i - 1 ? `Preset ${2 * i - 1}` : `Save Preset ${2 * i - 1}`}
+    </Button>
+    {#if i !== presetCount / 2 + 0.5}
+      <Button add="flex-1" on:click={() => { isDeleting ? deletePreset(2 * i) : updatePreset(2 * i); }} color="dark" light={active === 2 * i && !isDeleting}>
+        {isDeleting ? `Delete Preset ${2 * i}` : active !== 2 * i ? `Preset ${2 * i}` : `Save Preset ${2 * i}`}
+      </Button>
+    {/if}
```

Comment:
Instead of putting the buttons next to each other in js, you should make them next to each other with css. Like if the screen isn't wide, they can be on top of each other but as the space widens, the buttons start to form rows.

I have an example on [my website (change the window width and observe the project cards)](https://r2dev2.github.io/) with the css to do the effect [in the source code](https://github.com/r2dev2/r2dev2.github.io/blob/d7df0489e03276443dfb2e4ce8009a12719ee10b/src/App.svelte#L78). Doing something similar but with flex display could get you a bottom row that is completely full.

Reply:
I was gonna do that but I when i squished the screen past a certain point no other components shrunk like the import/export button so I never implemented it so that they would keep the same style as the import/export buttons.

Reply:
I'm just gonna switch it over from flexbox to a css grid quickly so that it cleans up the code so I don't need to explicitly do even and odd numbers for the time being and then I'll start working on automatically switching them over to a vertical mode.

Reply:
I think the whole UI needs to be redesigned so that it works with smaller screen sizes before I can do that because it overflows right now.